components/bridge/TxModal.tsx

```diff
@@ -91,123 +87,33 @@ const TxModal: React.FC<Props> = ({
-							? "DONE! YOU MAY NOW CLOSE THIS WINDOW."
-							: modalText}
-					</Heading>
-					{modalState !== "relayer" &&
-						modalState !== "bridgePaused" &&
-						modalState !== "error" &&
-						modalState !== "finished" && (
-							<Box sx={{ margin: "10px auto 50px" }}>
-								<CircularProgress size="3rem" sx={{ color: "black" }} />
-							</Box>
-						)}
-					{modalState === "relayer" && relayerStatus !== "Successful" && (
-						<Box sx={{ margin: "10px auto 20px" }}>
-							<CircularProgress size="3rem" sx={{ color: "black" }} />
-							<SmallText
-								sx={{
-									color: "black",
-									fontSize: "14",
-									margin: "10px auto 0",
-								}}
-							>
-								Confirmations: {confirms} / {eventConfirmations}
-							</SmallText>
-						</Box>
-					)}
-					{etherscanHash !== "" &&
-						etherscanHash !== "noTokenSelected" &&
-						modalState !== "relayer" && (
-							<Button
-								size="large"
-								variant="contained"
-								sx={{
-									backgroundColor: "primary.main",
-									width: "50%",
-									margin: "20px auto 50px",
-								}}
-							>
-								<Link
-									href={etherscanLink}
-									target="_blank"
-									rel="noopener noreferrer"
-								>
-									<Heading sx={{ color: "#FFFFFF", fontSize: "24px" }}>
-										View on Etherscan
-									</Heading>
-								</Link>
-							</Button>
-						)}
-					{(modalState === "bridgePaused" ||
-						modalState === "error" ||
-						modalState === "finished") && (
-						<Button
-							variant="contained"
-							sx={{
-								backgroundColor: "primary.main",
-								width: "50%",
-								margin: "50px auto 50px",
-							}}
-							onClick={() => resetModal()}
-						>
-							<Heading sx={{ color: "#FFFFFF", fontSize: "24px" }}>
-								close
-							</Heading>
-						</Button>
-					)}
-					{modalState === "relayer" && (
-						<Button
-							variant="contained"
-							sx={{
-								backgroundColor: "primary.main",
-								width: "50%",
-								margin: "50px auto 50px",
-							}}
-							onClick={() => {
-								updateRelayerStatus("");
-								resetModal();
-							}}
-							disabled={relayerStatus === "Successful" ? false : true}
-						>
-							<Heading sx={{ color: "#FFFFFF", fontSize: "24px" }}>
-								close
-							</Heading>
-						</Button>
-					)}
-				</StyledModal>
-			</Modal>
-		</Backdrop>
-	);
+	useEffect(() => {
+		if (modalOpen) {
+			if (
+				etherscanHash !== "" &&
+				etherscanHash !== "noTokenSelected" &&
+				modalState !== "relayer" &&
+				etherscanLink !== ""
+			) {
+				showDialog({
+					title: modalTitle,
+					message: modalText,
+					buttonText: "View on Etherscan",
+					link: etherscanLink,
+					disabled: false,
+				});
+			} else {
+				showDialog({
+					title: modalTitle,
+					message: modalText,
+					disabled: modalState === "withdrawCENNZside",
+					callback: resetModal,
+				});
+			}
+		}
+	}, [modalOpen, modalText, modalState, etherscanLink]);
```

Comment:
Missing quite a few dependencies here
![Screen Shot 2022-03-01 at 21 04 39](https://user-images.githubusercontent.com/97480229/156129429-6ff8c4b6-3ec1-4582-9306-04bccee1a5e3.png)

Reply:
added all modal atts to deps but disabled eslint for that line because was causing infinite loop with showdialog dep